app/src/main/java/com/example/internetshop/presentation/activity/fragments/BaseFragment.kt

```diff
@@ -7,45 +7,38 @@ import android.view.ViewGroup
 import android.widget.Toast
 import androidx.fragment.app.Fragment
 import androidx.lifecycle.ViewModelProvider
+import com.example.internetshop.R
 import com.example.internetshop.model.data.di.component.AppComponent
 import com.example.internetshop.presentation.InternetshopApplication
+import com.example.internetshop.presentation.activity.MainActivity
+import com.example.internetshop.presentation.viewModel.BottomNavViewModel
 import com.example.internetshop.presentation.viewModel.MultiViewModuleFactory
-import com.example.internetshop.presentation.viewModel.TitleViewModel
 import javax.inject.Inject
 
 abstract class BaseFragment : Fragment() {
 
     @Inject
     protected lateinit var factory: MultiViewModuleFactory
 
-    private val titleViewModel: TitleViewModel by lazy {
-        ViewModelProvider(requireActivity(), factory).get(TitleViewModel::class.java)
+    private val bottomNavViewModel: BottomNavViewModel by lazy {
+        ViewModelProvider(requireActivity(), factory).get(BottomNavViewModel::class.java)
     }
 
-    abstract fun inject(component: AppComponent)
-
-    abstract fun getTitle(): String
-
-    abstract fun getHomeVisibility(): Boolean
+    abstract fun setBottomNavVisibility(): Boolean
```

Comment:
As for me, this method shouldn't be abstract. Almost all fragments will have bottom nav visible. And only several one will haven't. Would be better to define this method like:
`fun setBottomNavVisibility(){
bottomNavViewModel.bottomNavVisibility.value = true
}

And override it in those fragments which don't need bottom nav to be visible, such as AuthentificationFragment

Reply:
fixed